README.md

```diff
@@ -145,17 +172,15 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
-
+https://github.com/CeeCeeLemons/cs3342_09/blob/master/state_machine_tests.pl
```

Comment:
Prolog? That's just plain pandering... :)

So, I really like what you've done, and it's a cool approach to writing a parser. But it doesn't really use a state machine.

Here's an answer that's closer to using a table:

~~~ prolog
valid_sentence(S) :- fsm(0, S).

fsm(0, [ the | Rest ])     :- fsm(1, Rest).
fsm(1, [ lazy | Rest ])    :- fsm(1, Rest).
fsm(1, [ smelly | Rest ])  :- fsm(1, Rest).
fsm(1, [ cat | Rest ])     :- fsm(2, Rest).
fsm(1, [ dog | Rest ])     :- fsm(2, Rest).
fsm(2, [ ate | Rest ])     :- fsm(3, Rest).
fsm(2, [ ran | Rest ])     :- fsm(3, Rest).
fsm(3, [ eoi ])            :- !.
fsm(3, [ slowly | Rest ])  :- fsm(4, Rest).
fsm(3, [ noisily | Rest ]) :- fsm(4, Rest).
fsm(4, [ eoi ])            :- !.
~~~
So, I'm grading this as +3 bonus for the Prolog, which made me happy, and -6 for not using a table

# -3